x-pack/filebeat/input/azureblobstorage/input.go

```diff
@@ -0,0 +1,150 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+// +build !aix
+
+package azureblobstorage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	cursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	"github.com/elastic/beats/v7/libbeat/feature"
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/azureblobstorage/state"
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/azureblobstorage/types"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type azurebsInput struct {
+	config     config
+	serviceURL string
+}
+
+const (
+	inputName string = "azureblobstorage"
+)
+
+func Plugin(log *logp.Logger, store cursor.StateStore) v2.Plugin {
+	return v2.Plugin{
+		Name:       inputName,
+		Stability:  feature.Experimental,
+		Deprecated: false,
+		Info:       "Azure Blob Storage logs",
+		Doc:        "Collect logs from Azure Blob Storage Service",
+		Manager: &cursor.InputManager{
+			Logger:     log,
+			StateStore: store,
+			Type:       inputName,
+			Configure:  configure,
+		},
+	}
+}
+
+func configure(cfg *conf.C) ([]cursor.Source, cursor.Input, error) {
+	config := defaultConfig()
+	if err := cfg.Unpack(&config); err != nil {
+		return nil, nil, err
+	}
+
+	var sources []cursor.Source
+	for _, c := range config.Containers {
+		container := tryOverrideOrDefault(config, c)
+		sources = append(sources, &types.Source{
+			AccountName:   config.AccountName,
+			ContainerName: c.Name,
+			MaxWorkers:    *container.MaxWorkers,
+			Poll:          *container.Poll,
+			PollInterval:  *container.PollInterval,
+		})
+	}
+
+	url := fmt.Sprintf("https://%s.blob.core.windows.net/", config.AccountName)
```

Comment:
```suggestion
	url := "https://" + config.AccountName + ".s.blob.core.windows.net/"
```

Reply:
The URL should also be fully configurable for testing purposes. I imagine we will want to integration test with their emulator as well as use this in Fleet testing.

https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=docker-hub

Reply:
@andrewkroh Sure will make this fully configurable, and I guess since **fmt.Sprintf** uses reflection internally , it's more optimal to use a simple concat since everything of the same type ?